django/views/generic/dates.py

```diff
@@ -696,7 +699,7 @@ def _get_next_prev(generic_view, date, is_previous, period):
         # Snag the first object from the queryset; if it doesn't exist that
         # means there's no next/previous link available.
         try:
-            result = getattr(qs[0], date_field)
+            result = functools.reduce(getattr, date_field.split(LOOKUP_SEP), qs[0])
```

Comment:
The use of `reduce` here not obvious. I think it needs a comment, something like: 

    # Walk parts of `date_field` returning last element. 

What happens here if `date_field` includes an incorrect part, e.g. `artist__not_there`? It's not going to raise an `IndexError` in that case. Do we handle this at all?